components/browser/engine-gecko-nightly/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineSessionTest.kt

```diff
@@ -1902,15 +1902,58 @@ class GeckoEngineSessionTest {
     }
 
     @Test
-    fun `Handle new window load requests`() {
```

Comment:
The original test was testing load requests, now the new test is "just" covering the app link interceptor. Let's also cover the case where we don't intercept?

Reply:
Sure, I'll add test cases where we don't intercept.